chromium_src/chrome/browser/search_engines/template_url_service_factory.cc

```diff
@@ -5,13 +5,11 @@
 
 #include "chrome/browser/search_engines/template_url_service_factory.h"
 
-#include "chrome/browser/history/history_service_factory.h"
-#include "chrome/browser/profiles/incognito_helpers.h"
-#include "chrome/browser/webdata_services/web_data_service_factory.h"
+#define WithAshInternals(...)                      \
+  WithAshInternals(__VA_ARGS__)                    \
+      .WithRegular(ProfileSelection::kOwnInstance) \
```

Comment:
this is covered with tests in `//brave/browser/search_engines/search_engine_provider_service_browsertest.cc`.